google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/AddCombinedRuleUserList.java

```diff
@@ -0,0 +1,175 @@
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * Creates a rule-based user list for people that have visited two different sections of a website.
+ */
+public class AddCombinedRuleUserList {
+
+  private static class AddCombinedRuleUserListParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+  }
+
+  public static void main(String[] args) {
+    AddCombinedRuleUserListParams params = new AddCombinedRuleUserListParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new AddCombinedRuleUserList().runExample(googleAdsClient, params.customerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+    String urlString = "url__";
+
+    // Creates a rule targeting any user that visited a url that equals 'example.com/example1'.
+    UserListRuleItemInfo rule1 =
+        UserListRuleItemInfo.newBuilder()
+            // Uses a built-in parameter to create a domain URL rule.
+            .setName(StringValue.of(urlString))
+            .setStringRuleItem(
+                UserListStringRuleItemInfo.newBuilder()
+                    .setOperator(UserListStringRuleItemOperator.EQUALS)
+                    .setValue(StringValue.of("example.com/example1"))
```

Comment:
Is it OK that this doesn't include the protocol or `www.` prefix?

Reply:
Good catch. Per https://support.google.com/google-ads/answer/6297497?hl=en, this example does require the protocol. Adding the `www. prefix` isn't necessary, per se, it just defines a different url. I've added the protocol to the example.